src/canvas-view/canvas/canvas-directive.js

```diff
@@ -0,0 +1,440 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+  .filter('trustAsResourceUrl', ['$sce', function ($sce) {
+    return function (val) {
+      return $sce.trustAsResourceUrl(val);
+    };
+  }])
+
+  //
+  // Directive that generates the rendered chart from the data model.
+  //
+  .directive('pfCanvas', function ($document) {
+    return {
```

Comment:
I believe the requirement for the 4.x release was to use components. We converted all the directives to components and don't think we should, at this point, introduce new directives.